pkg/controllers/dynakube/connectioninfo/oneagent/conditions.go

```diff
@@ -0,0 +1,22 @@
+package oaconnectioninfo
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	oaConnectionInfoConditionType = "OneAgentConnectionInfo"
+
+	EmptyCommunicationHostsReason = "EmptyCommunicationHosts"
+)
+
+func setEmptyCommunicationHostsCondition(conditions *[]metav1.Condition) {
```

Comment:
wdyt if we sligthly change the way we set conditions?

Instead of having bunch of setXXX condition? we will have just fabric of condition and then use 

`meta.SetStatusCondition(r.dynakube.Conditions(), NewEmptyCommunicationHostsCondition())`
?

Reply:
why, because in some cases you have type as argument, sometimes no.

Reply:
For common conditions:
```
// I prefer this one, as its shorter 
conditions.SetKubeApiErrorCondition(r.dynakube.Conditions(), oaConditionType, err)
```
vs
```
meta.SetStatusCondition(r.dynakube.Conditions(), conditions.NewKubeApiError(oaConditionType, err))
```

For package specific conditions:
```
// I prefer this one, as its shorter and looks more similar to the "common conditions" scenario
setEmptyCommunicationHostsCondition(r.dynakube.Conditions())
```
vs
```
meta.SetStatusCondition(r.dynakube.Conditions(), newEmptyCommunicationHostsCondition())
```